python/ray/data/impl/stats.py

```diff
@@ -110,16 +111,27 @@ def get(self, stats_uuid):
         )
 
 
-# Actor handle, job id the actor was created for.
-_stats_actor = [None, None]
+# Actor handle, job id, client id the actor was created for.
+_stats_actor = [None, None, None]
 
 
 def _get_or_create_stats_actor():
```

Comment:
Hmm I don't remember why we did it this way, would it be better to use `actor.options(get_if_exists=True, name="_dataset_stats_actor").remote()` instead to get-or-create the actor?

Reply:
I thought about changing to this option (getting by name) as well, but it seems not meet the requirement. 
So the requirement here is that this actor is reused across datasets created by this same process. For the getting by name approach:
- It may work for the same dataset, but after it's completed and the actor refcount goes to zero, the datasets created later will not be able to get that same actor (have to recreate).
- Alternatively, we may create a detached actor, but in that case it will be shared even across different driver processes.

Keeping a reference here and clearing it upon new connection or new driver is essentially what this PR does to serve this requirement.

Reply:
Hm, sharing across drivers should be ok right? That should be uncommon case anyway.

Reply:
If that's not a concern, then getting by name is indeed simpler.

Reply:
So this stats actor will never be cleaned up for the lifetime of the Ray cluster, and the read stats for each `Dataset` will never get cleaned up within the actor, which is a bit of a leaky lifecycle. This seems fine for now to unblock the Ray Client use case, but we should probably open a P2 to improve the stats actor lifecycle, or eliminate the stats actor all together, if possible.

Reply:
My previous code was made quite close to the existing way. I think the feedback here is it's a bit complicated AND detached actor isn't a concern to use. It may make sense to have a new kind of lifetime between refcounted and detached, e.g. per-job lifetime actor.